File: 2024_java_basics/14-nested-loops-lab/tasks/02-equal-sums-even-odd-position.java
import java.util.Scanner;

public class EqualSumsEvenOddPosition {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int n1 = scanner.nextInt();
        int n2 = scanner.nextInt();
        
        for (int i = n1; i <= n2; i++) {
            String number = Integer.toString(i);
            int evenSum = 0;
            int oddSum = 0;
            
            for (int j = 0; j < number.length(); j++) {
                int digit = Character.getNumericValue(number.charAt(j));
                
                if (j % 2 == 0) {
                    evenSum += digit;
                } else {
                    oddSum += digit;
                }
            }
            
            if (evenSum == oddSum) {
                System.out.print(i + " ");
            }
        }
        
        scanner.close();
    }
}

File: 2022_c#_basics/14-nested-loops-exercise/02-equal-sums-even-odd-position.cs
using System;

namespace _02_Equal_Sums_Even_Odd_Position
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int n1 = int.Parse(Console.ReadLine());
            int n2 = int.Parse(Console.ReadLine());

            for (int i = n1; i <= n2; i++)
            {
                string number = i.ToString();
                int evenSum = 0;
                int oddSum = 0;
                
                for (int j = 0; j < number.Length; j++)
                {
                    int digit = int.Parse(number[j].ToString());

                    if (j % 2 == 0)
                    {
                        evenSum += digit;
                    }
                    else
                    {
                        oddSum += digit;
                    }
                }

                if (evenSum == oddSum)
                {
                    Console.Write($"{i} ");
                }
                
            }

            
        }
    }
}
